docs/changes/DM-45938.feature.rst

```diff
@@ -0,0 +1,2 @@
+Added automatic ID generation for objects in Felis schemas when the `--id-generation` flag is included on the command line.
+This is supported for the `create` and `validate` commands.
```

Comment:
Please add here that the change also "add[s] validator that checks if index names are unique within the schema"